helix/transforms/tigress/tigress.py

```diff
@@ -0,0 +1,166 @@
+import json
+import magic
+import os
+import shutil
+from urllib import parse, request as req
+
+from json.decoder import JSONDecodeError
+
+from ... import utils
+from ... import transform
+from ... import exceptions
+
+
+class TigressError(Exception):
+    """Raised when Tigress fails."""
+
+    pass
+
+
+class TigressDependency(utils.Dependency):
+    """Using the Tigress Transform"""
+
+    def __init__(self, name):
+        if os.name != "posix":
+            raise exceptions.ConfigurationError(
+                "unsupported platform for this dependency type: {}".format(os.name)
+            )
+
+            self.name = name
+
+    def install(self, verbose):
+        """Downloading Tigress binaries and adding execution permission."""
+        print("  By installing, you accept the Tigress End-User License Agreement.")
+        url = "http://tigress.cs.arizona.edu/cgi-bin/projects/tigress/download.cgi"
+        data = {
+            "accept": "Accept and Download",
+            "mode": "download",
+            "buffer": "address:----email:----file:tigress-3.1-bin.zip----name:----remote_addr=----timestamp:----",
+            "file": "tigress-3.1-bin.zip",
+            "destfile": "tigress-3.1-bin.zip",
+        }
+        data = parse.urlencode(data).encode()
+        request = req.Request(url=url, data=data)
+        response = req.urlopen(request)
+
+        destination = os.path.expanduser("~/bin")
+        if not os.path.exists(destination):
+            os.makedirs(destination)
+
+        temp = destination + "/tigress-3.1-bin.zip"
+
+        open(temp, "wb").write(response.read())
+        shutil.unpack_archive(temp, destination)
+        os.remove(temp)
+
+        exec_permission = "chmod -R u+x " + destination + "/tigress"
+        utils.run(exec_permission, propagate=True)
```

Comment:
Does the entire directory need execute permissions? Probably best to just modify a single binary if possible. You could also use https://docs.python.org/3/library/os.html#os.chmod here instead of invoking a shell command

Reply:
There are several binaries and scripts with execute permissions. A proposed solution is to try to preserve all the permissions of the files in the download. 

Since `zipfile.extractall()` and `shutil.unpack_archive()` don't  preserve permissions there's a workaround that overrides the `_extract_member` method used by `zipfile.extractall()` and adds the original permissions to the extracted files ([workaround and reference](https://github.com/helix-datasets/helix/blob/2a1fd020f8a4997e72849e7444efcfbcc847a6dc/helix/transforms/tigress/utils.py#L45-L65), [library source](https://github.com/python/cpython/blob/b3bd69c1bfb431fe75cc301f97494f9f98df2a19/Lib/zipfile.py#L1711-L1754)). It also uses `os.chmod` instead of a shell command.